### PR TITLE
Loans

### PR DESCRIPTION
**Pull request by HudokonNau**

### CODE REVIEW

File: App.Common/App.Common.csproj [hunk 1]
```diff
@@ -6,7 +6,9 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Castle.Core" Version="4.4.0" />
```
Comment (ghost) [inline App.Common/App.Common.csproj]: What was the reason of this changes ?

File: App.Loans/Controllers/LoansController.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+using App.Loans.Services;
+
+namespace App.Loans.Controllers
+{
+    /// <summary>
+    /// This is example controller
+    /// IMPORTANT the route to your won module should be 'api/{yourModuleName}' in order to avoid conflicts with other modules
```
Comment (ghost) [inline App.Loans/Controllers/LoansController.cs]: This comments are related to the example controller and are extra here. Please, remove them, or modify to correspond to your current controller

File: App.Loans/Controllers/LoansController.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            _validateService = validateService;
+            _logger = logger;
+            _valuesManager = valuesManager;
+        }
+
+        // GET api/loans/All
+        [HttpGet("All")]
+        public ActionResult<IEnumerable<string>> Get()
+        {
+            _logger.LogInformation("NOTHING");
```
Comment (ghost) [inline App.Loans/Controllers/LoansController.cs]: this is extra for now

File: App.Loans/Models/Loan.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace App.Loans.Models
+{
+    public class Loan
+    {
+        private double money;//денежки
+        public int time { get; set; }//кол-во платежей (ежемесячных)
+        public double money_taked { get; set; }//скок взял
+        public double money_back { get; set; }//скок вернул
+        public double money_left { get { return money; } set { money = money_taked + (money_taked * (time/12) *percent) - money_back; } } //скок осталось вернуть
```
Comment (ghost) [inline App.Loans/Models/Loan.cs]: 1) you use the same calculation formula in 2 places. You repeat yourself, that means you should this logic into distinct method and reuse it.
2) "money_left" property should be readonly and should not have "set" modifier, because it is calculated from values of "money_taked" and "money_back"

File: App.Loans/Repositories/LoansRepository.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+using App.Loans.Models;
+using App.Configuration;
+using App.Repositories;
+
+namespace App.Loans.Repositories
+{
+    public interface ILoansRepository
+    {
+        IEnumerable<Loan> GetAll();
+        IEnumerable<string> GetValues();
```
Comment (ghost) [inline App.Loans/Repositories/LoansRepository.cs]: What this interface method did ? Is it really needed, because you already have GetAll method ?

File: App.Loans/Repositories/LoansRepository.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+        Loan Get(int id);
+    }
+    public class LoansRepository : ILoansRepository, ITransientDependency
+    {
+        private Loan[] loans = { new Loan(10000, 10, 24), new Loan(28860, 3, 6), new Loan(5000, 0.01, 3) };
```
Comment (ghost) [inline App.Loans/Repositories/LoansRepository.cs]: Please, place each single statement on new line. It will make code much easier to read

File: App.Loans/Services/ValidateServices.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+using App.Loans.Models;
+
+namespace App.Loans
+{
+    public class ValidateServices : IValidateServices
+    {
+        public void ValidateLoan(Loan loan)
+        {
+            if (loan == null)
+                throw new Exception("Loan doesn't exist");
```
Comment (ghost) [inline App.Loans/Services/ValidateServices.cs]: This is not the responsibility of Validator. Null checking is the responsibility of caller method

**Review by ghost — CHANGES_REQUESTED** [review]
In c# it is common to use PascalCase style for methods naming. Almost all interfaces in this PR violates this rule. Please, re-read articles I attached to laboratory work description (especially about methods naming in C#) and fix all snake_case naming

File: App.Loans/Repositories/LoansRepository.cs (hunk 5 above)
Comment (HudokonNau) [reply App.Loans/Repositories/LoansRepository.cs]: Yes i need it for resolve the task.

File: App.Common/App.Common.csproj (hunk 1 above)
Comment (ghost) [reply App.Common/App.Common.csproj]: Why this comment was resolved ? You did not answered

File: App.Loans/Repositories/LoansRepository.cs (hunk 5 above)
Comment (ghost) [reply App.Loans/Repositories/LoansRepository.cs]: Not a good explanation

File: App.TestModule/App.Example.csproj [hunk 8]
```diff
@@ -5,11 +5,14 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Castle.Core" Version="4.4.0" />
```
Comment (ghost) [inline App.TestModule/App.Example.csproj]: Why you have changed this module ?

File: App.TestModule/ValuesManager.cs [hunk 9]
```diff
@@ -1,7 +1,6 @@
 ﻿using System.Collections.Generic;
 using App.Configuration;
 using App.Repositories;
-
```
Comment (ghost) [inline App.TestModule/ValuesManager.cs:4]: what is the reason of this change ?

**Review by ghost — CHANGES_REQUESTED** [review]

File: App.Common/App.Common.csproj (hunk 1 above)
Comment (HudokonNau) [reply App.Common/App.Common.csproj]: maybe I connected an extra package when setting up nuget.

File: App.Loans/Repositories/LoansRepository.cs (hunk 5 above)
Comment (HudokonNau) [reply App.Loans/Repositories/LoansRepository.cs]: I need it to use in controller this method. 
This method returns information(Values) about active loans. Like I need in the task.

File: App.TestModule/ValuesManager.cs (hunk 9 above)
Comment (HudokonNau) [reply App.TestModule/ValuesManager.cs:4]: Changes with hidden meaning

File: App.TestModule/App.Example.csproj (hunk 8 above)
Comment (HudokonNau) [reply App.TestModule/App.Example.csproj]: I just manage nuget packages

Comment (HudokonNau) [reply App.TestModule/App.Example.csproj]: And connect this , because I have it in button restore nuget packages

**Review by ghost — APPROVED** [review]

**Comment by ghost** [comment]
Please, resolve conflicts
